controllers/vaultpkisecret_controller.go

```diff
@@ -229,61 +218,89 @@ func (r *VaultPKISecretReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 		}
 		data[corev1.TLSPrivateKeyKey] = data["private_key"]
 	}
+
+	if b, err := json.Marshal(data); err == nil {
+		newMAC, err := r.HMACValidator.HMAC(ctx, r.Client, b)
+		if err != nil {
+			logger.Error(err, "HMAC data")
+			o.Status.Error = consts.ReasonHMACDataError
+			if err := r.updateStatus(ctx, o); err != nil {
+				return ctrl.Result{}, err
+			}
+			return ctrl.Result{
+				RequeueAfter: computeHorizonWithJitter(requeueDurationOnError),
+			}, nil
+		}
+		o.Status.SecretMAC = base64.StdEncoding.EncodeToString(newMAC)
+	}
+
 	if err := helpers.SyncSecret(ctx, r.Client, o, data); err != nil {
-		return ctrl.Result{}, err
+		logger.Error(err, "Sync secret")
+		o.Status.Error = consts.ReasonSecretSyncError
+		if err := r.updateStatus(ctx, o); err != nil {
+			return ctrl.Result{}, err
+		}
+		return ctrl.Result{
+			RequeueAfter: computeHorizonWithJitter(requeueDurationOnError),
+		}, nil
 	}
 
 	reason := consts.ReasonSecretSynced
-	if timeToRenew {
+	if o.Status.SerialNumber != "" {
 		reason = consts.ReasonSecretRotated
 		// rollout-restart errors are not retryable
 		// all error reporting is handled by helpers.HandleRolloutRestarts
 		_ = helpers.HandleRolloutRestarts(ctx, r.Client, o, r.Recorder)
 	}
 
 	// revoke the certificate on renewal
-	if o.Spec.Revoke && timeToRenew && o.Status.SerialNumber != "" {
+	if o.Spec.Revoke && o.Status.SerialNumber != "" {
 		if err := r.revokeCertificate(ctx, logger, o); err != nil {
-			return ctrl.Result{}, err
+			logger.Error(err, "Certificate revocation")
+			o.Status.Error = consts.ReasonCertificateRevocationError
+			return ctrl.Result{
+				RequeueAfter: computeHorizonWithJitter(requeueDurationOnError),
+			}, nil
 		}
 	}
 
 	o.Status.Valid = true
 	o.Status.Error = ""
 	o.Status.SerialNumber = certResp.SerialNumber
 	o.Status.Expiration = certResp.Expiration
+	o.Status.LastRotation = time.Now().Unix()
+	o.Status.LastGeneration = o.GetGeneration()
 	if err := r.updateStatus(ctx, o); err != nil {
 		logger.Error(err, "Failed to update the status")
 		return ctrl.Result{}, err
 	}
 
-	if err := r.addFinalizer(ctx, logger, o); err != nil {
-		return ctrl.Result{}, err
-	}
-
 	logger.Info("Successfully updated the secret")
 	r.recordEvent(o, reason, "Secret synced")
 
+	horizon, _ := computePKIRenewalWindow(ctx, o, .05)
+	logger.Info("Requeue", "horizon", horizon)
```

Comment:
This might be good to log at debug level.

Reply:
Dropped it with  d7a5ad7c97a73d39d8f253fca0ae8c3f4965d8d0